File: ecommerce_manager_web/src/main/webapp/js/service/itemCatService.js
//服务层
app.service('itemCatService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../itemCat/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../itemCat/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../itemCat/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../itemCat/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../itemCat/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../itemCat/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../itemCat/search.do?page='+page+"&rows="+rows, searchEntity);
	}

	//根据上级id，返回全部下级list
	this.findByParentId=function (parentId) {
		return $http.get('../itemCat/findByParentId.do?parentId='+parentId);
	}
	//根据上级id，返回当前页下级list
	this.findByParentIdPage=function (parentId,page,rows) {
		return $http.get('../itemCat/findByParentIdPage.do?parentId='+parentId+'&page='+page+'&rows='+rows);
	}
	//根据上级id，搜索
	this.searchByParentId=function(parentId,page,rows,searchEntity){
		return $http.post('../itemCat/searchByParentId.do?parentId='+parentId+'&page='+page+'&rows='+rows, searchEntity);
	}

});

File: ecommerce_goods_interface/src/main/java/goods/service/ItemCatService.java
package goods.service;

import java.util.List;

import pojo.TbItemCat;

import entity.PageResult;

/**
 * 服务层接口
 */
public interface ItemCatService {

    /**
     * 返回全部列表
     *
     * @return
     */
    public List<TbItemCat> findAll();

    /**
     * 返回分页列表
     *
     * @return
     */
    public PageResult findPage(int pageNum, int pageSize);


    /**
     * 增加
     */
    public void add(TbItemCat itemCat);


    /**
     * 修改
     */
    public void update(TbItemCat itemCat);


    /**
     * 根据ID获取实体
     *
     * @param id
     * @return
     */
    public TbItemCat findOne(Long id);


    /**
     * 批量删除
     *
     * @param ids
     */
    public void delete(Long[] ids);

    /**
     * 分页
     *
     * @param pageNum  当前页 码
     * @param pageSize 每页记录数
     * @return
     */
    public PageResult findPage(TbItemCat itemCat, int pageNum, int pageSize);


    /**
     * 根据上级ID，返回全部下级列表
     *
     * @param parentId
     * @return
     */
    public List<TbItemCat> findByParentId(Long parentId);

    /**
     * 根据上级ID，返回当前页下级列表
     * @param parentId
     * @param pageNum
     * @param pageSize
     * @return
     */
    public PageResult findByParentId(Long parentId,int pageNum, int pageSize);

    /**
     * 查询下级+分页
     * @param itemCat
     * @param parentId
     * @param pageNum  当前页 码
     * @param pageSize 每页记录数
     * @return
     */
    public PageResult findPageByParentId(TbItemCat itemCat, Long parentId, int pageNum, int pageSize);

}
